plugins/services/src/js/components/DeploymentsModal.js

```diff
@@ -483,7 +483,7 @@ class DeploymentsModal extends mixin(StoreMixin) {
       {});
     }
 
-    let statusText = !item.isStale ? item.getStatus() : null;
+    let statusText = !item.isStale && item.getStatus ? item.getStatus() : null;
```

Comment:
Do you think it would be overkill to check `typeof item.getStatus === "function"`?

Reply:
I also considered doing `item.isStale === false` b/c if the struct has `isStale` it *should* also have `getStatus()`, but it felt less explicit and could be confusing looking at that code sometime down the road.